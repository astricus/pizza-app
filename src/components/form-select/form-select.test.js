import React from 'react';
import { shallow } from 'enzyme';

import FormSelect from './form-select.component';

describe('FormSelect component', () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: 'currency',
      handleChange: mockHandleChange,
      value: 'USD',
    };

    wrapper = shallow(<FormSelect {...mockProps} />);
  });

  it('should render FormSelect component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange method when input changes', () => {
    wrapper.find('FormSelectContainer').simulate('change');

    expect(mockHandleChange).toHaveBeenCalled();
  });
});
