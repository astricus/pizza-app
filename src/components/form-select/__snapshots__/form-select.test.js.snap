// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect component should render FormSelect component 1`] = `
<styled.div>
  <FormSelectContainer
    onChange={[MockFunction]}
    value="USD"
  >
    <option
      value="USD"
    >
      USD
    </option>
    <option
      value="EUR"
    >
      EUR
    </option>
  </FormSelectContainer>
  <FormSelectLabel
    className="shrink"
  >
    currency
  </FormSelectLabel>
</styled.div>
`;
